### PR TITLE
Simplify deployments and update to new production buckets

### PR DESCRIPTION
**Pull request by mramato**
1. Remove all custom AWS code and replace with aws cli usage
2. No longer gzip everything manually, let AWS compress automatically for us.
3. Remove no longer needed npm modules

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @mramato!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/update-deployment/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by mramato** [comment]
Just realized I missed a few things, will bump this PR when ready for review.

**Comment by mramato** [comment]
@ggetz this is ready.

I mainly matched what was already there but I think this will make it way easier to change going forward. I think there is a conversation to be had over exactly what we should deploy for CI and what we shouldn't, but I didn't want to expand the scope of this PR.

I also noticed the deployment/zip links are broken across all PRs (I'll see if I can put a small change for that and open a separate PR for it). If you fix the links manually based on the above checks, they are all correct and work.

**Comment by mramato** [comment]
> I also noticed the deployment/zip links are broken across all PRs (I'll see if I can put a small change for that and open a separate PR for it). If you fix the links manually based on the above checks, they are all correct and work.

This turned out to be a 1-liner so I just pushed it here. See https://github.com/CesiumGS/cesium/pull/11649/commits/57df63e7fb05b80e8ffb91a2c206d4c467db1aa0

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -97,7 +97,21 @@ jobs:
...
+        run: |
+          aws s3 sync . s3://cesium-public-builds/cesium/$BRANCH/ \
+          --cache-control "no-cache" \
+          --exclude ".git/*" \
+          --exclude ".concierge/*" \
+          --exclude ".github/*" \
+          --exclude ".husky/*" \
+          --exclude ".vscode/*" \
+          --exclude "Build/Coverage/*" \
+          --exclude "Build/CesiumDev/*" \
```
Comment (ggetz) [inline .github/workflows/dev.yml:109]: Lets also exclude `Build/Specs/e2e` to save space.

File: .github/workflows/prod.yml [hunk 2]
```diff
@@ -45,4 +45,9 @@ jobs:
         run: npm run build-apps
       - name: deploy to cesium.com
         if: ${{ env.AWS_ACCESS_KEY_ID != '' }}
-        run: npm run deploy-s3 -- -b "cesium.com-next" -c 'public, max-age=1800' --skip --confirm
\ No newline at end of file
+        run: |
+          unzip Cesium-$(cat package.json | jq -r '.version').zip -d Build/release/ -x "Apps"
```
Comment (ggetz) [inline .github/workflows/prod.yml:49]: Is `jq` built in or does it need to be installed?

File: gulpfile.js [hunk 3]
```diff
@@ -22,21 +19,9 @@ import mergeStream from "merge-stream";
...
 import yargs from "yargs";
-import {
-  S3Client,
-  DeleteObjectsCommand,
-  HeadObjectCommand,
-  ListObjectsCommand,
-  PutObjectCommand,
-} from "@aws-sdk/client-s3";
-import { Upload } from "@aws-sdk/lib-storage";
-import mime from "mime";
```
[the remark this answers is not in the document]
Comment (mramato) [reply gulpfile.js:33]: Looks like that's the case. Good catch.

File: .github/workflows/dev.yml (hunk 1 above)
Comment (mramato) [reply .github/workflows/dev.yml:109]: Added

**Comment by mramato** [comment]
@ggetz updated and back to you.

**Comment by ggetz** [comment]
Great, thanks @mramato! I will merge once CI finishes.

**Pull request merged**